[2.13beta] _deckNewLimitSingle stackoverflow

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/663437b4-40fc-44c2-9061-6decb6890c1f

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:399)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.StackOverflowError: stack size 1039KB
at com.ichi2.utils.JSONObject.getInt(JSONObject.java:1)
at com.ichi2.libanki.sched.SchedV2._deckNewLimitSingle(SchedV2.java:1)
at com.ichi2.libanki.sched.SchedV2.a(SchedV2.java:1)
at com.ichi2.libanki.sched.f.operation(Unknown Source:2)
at com.ichi2.libanki.sched.SchedV2._deckNewLimit(SchedV2.java:6)
at com.ichi2.libanki.sched.SchedV2._deckNewLimit(SchedV2.java:1)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:6)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:6)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:6)
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java
@@ -731,7 +731,7 @@ public class Sched extends SchedV2 {
         // Since we didn't get a card and the count is non-zero, we
         // need to check again for any cards that were removed from
         // the queue but not buried
-        _resetRev(mCurrentCard);
+        _resetRev();
         return _fillRev(true);
     }
 

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
@@ -719,14 +719,6 @@ public class SchedV2 extends AbstractSched {
      * New cards **************************************************************** *******************************
      */
 
-    /** Same as _resetNew, but assume discardCard is currently in the reviewer and so don't conunt it.*/
-    protected void _resetNew(@Nullable Card discardCard) {
-        _resetNew();
-        if (discardCard != null && discardCard.getQueue() == Consts.QUEUE_TYPE_NEW) {
-            mNewCount--;
-        }
-    }
-
     protected void _resetNewCount() {
         mNewCount = _walkingCount((Deck g) -> _deckNewLimitSingle(g),
                                   (long did, int lim) -> _cntFnNew(did, lim));
@@ -836,7 +828,7 @@ public class SchedV2 extends AbstractSched {
         // if we didn't get a card, since the count is non-zero, we
         // need to check again for any cards that were removed
         // from the queue but not buried
-        _resetNew(mCurrentCard);
+        _resetNew();
         return _fillNew(true);
     }
 
@@ -1518,15 +1510,6 @@ public class SchedV2 extends AbstractSched {
                                         mToday, lim);
     }
 
-
-    /** Same as _resetRev, but assume discardCard is currently in the reviewer and so don't conunt it.*/
-    protected void _resetRev(@Nullable Card discardCard) {
-        _resetRev();
-        if (discardCard != null && discardCard.getQueue() == Consts.QUEUE_TYPE_REV) {
-            mRevCount--;
-        }
-    }
-
     // Overriden: V1 uses _walkingCount
     protected void _resetRevCount() {
         int lim = _currentRevLimit();
@@ -1598,7 +1581,7 @@ public class SchedV2 extends AbstractSched {
         // since we didn't get a card and the count is non-zero, we
         // need to check again for any cards that were removed from
         // the queue but not buried
-        _resetRev(mCurrentCard);
+        _resetRev();
         return _fillRev(true);
     }
 

--- a/AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/libanki/sched/AbstractSchedTest.java
@@ -27,6 +27,7 @@ import com.ichi2.libanki.Decks;
 import com.ichi2.libanki.Model;
 import com.ichi2.libanki.Models;
 import com.ichi2.libanki.Note;
+import com.ichi2.testutils.AnkiAssert;
 import com.ichi2.utils.JSONArray;
 
 import org.junit.Before;
@@ -376,5 +377,25 @@ mw.col.sched.extendLimits(1, 0)
 
         boolean hasMatch = decks.all().stream().anyMatch(x -> name.equals(x.getString("name")));
         assertThat(String.format("Deck %s should exist", name), hasMatch, is(true));
+    }
+
+
+
+    @Test
+    public void regression_7066() {
+        Collection col = getCol();
+        DeckConfig dconf = col.getDecks().getConf(1);
+        dconf.getJSONObject("new").put("bury", true);
+        AbstractSched sched = col.getSched();
+        addNoteUsingBasicAndReversedModel("foo", "bar");
+        addNoteUsingBasicModel("plop", "foo");
+        col.reset();
+        Card card = sched.getCard();
+        sched.setCurrentCard(card);
+        sched.preloadNextCard();
+        sched.answerCard(card, Consts.BUTTON_THREE);
+        card = sched.getCard();
+        sched.setCurrentCard(card);
+        AnkiAssert.assertDoesNotThrow(() -> sched.preloadNextCard());
     }
 }

